libr/asm/arch/avr/format.c

```diff
@@ -156,10 +156,34 @@ static int formatDisassembledOperand(avrDisassembleContext *context, char *strOp
 			fOptions.addressFieldWidth, context->longAddress);
 		break;
 	case OPERAND_IO_REGISTER:
-		retVal = sprintf(strOperand, "%s%02x",
-			OPERAND_PREFIX_IO_REGISTER,
-			dInstruction.operands[operandNum]);
+	{
+		char *current_register = NULL;
+		int is_register_found = 0;
+		switch(dInstruction.operands[operandNum])
+		{
+		    case 0x3f:
+		        current_register = "sreg\0";
+		        break;
+		    case 0x04:
+		        current_register = "adcl\0";
+		        break;
+		    case 0x05:
+		        current_register = "adch\0";
+		        break;
+		    default:
+		        is_register_found = -1;
+		        retVal = sprintf(strOperand, "0x%x", dInstruction.operands[operandNum]);
```

Comment:
```suggestion
		        retVal = snprintf (strOperand, sizeof (strOperand), "0x%x", dInstruction.operands[operandNum]);
```

Reply:
It raises an error:

/home/ME/radare2/libr/..//libr/anal/p/../../asm/arch/avr/format.c: In function ‘formatDisassembledOperand’:
/home/ME/radare2/libr/..//libr/anal/p/../../asm/arch/avr/format.c:179:46: warning: argument to ‘sizeof’ in ‘snprintf’ call is the same expression as the destination; did you mean to provide an explicit length? [-Wsizeof-pointer-memaccess]
  179 |        retVal = snprintf (strOperand, sizeof (strOperand), "0x%x", dInstruction.operands[operandNum]);
      |                                              ^


and strlen () returns alway 1. Not enough to print the register.

Reply:
oh i saw strOperand[256]; somewhere.. but this is char *, so you need to pass the size of this buffer in the caller to know its size

Reply:
strlen(strOperand) returns 1